staging/src/k8s.io/apimachinery/pkg/runtime/interfaces.go

```diff
@@ -256,6 +279,25 @@ type Object interface {
 	DeepCopyObject() Object
 }
 
+// CacheableObject allows an object to cache its different serializations
+// to avoid performing the same serialization multiple times.
+type CacheableObject interface {
+	// CacheEncode writes an object to a stream. The <encode> function will
+	// be used in case of cache miss. If CacheableObject is a wrapper, then
+	// the wrapped object should be passed to <encode> function.
```

Comment:
clarify a *deep copy* of the wrapped object is passed to `encode`

Reply:
done